server.py

```diff
@@ -22,20 +22,29 @@
 
 app = Flask(__name__)
 
-class DoorOpener(object):
+class DoorManager(object):
     def __init__(self):
         self.open_door_ts = 0.0
         self.open_cond = threading.Condition()
+        self.locked = True
 
     def open(self):
         with self.open_cond:
             self.open_door_ts = time.time()
             logging.info("Set open_door_ts=%s", self.open_door_ts)
             self.open_cond.notify()
 
+    def unlock(self):
+        logging.info('Unlocking door.')
+        self.locked = False
+
+    def lock(self):
+        logging.info('Locking door.')
+        self.locked = True
```

Comment:
correctness nit: This variable (`self.locked`) is accessed from more than one thread at once, and needs to be locked with the `self.open_cond` internal lock.